src/Poseidon/RagnarokServer.pm

```diff
@@ -845,37 +879,64 @@ sub SendCharacterList
 	my $data;
 	if ($self->{type}->{$config{server_type}}->{charListPacket} eq '0x82d') {
 		$data = $accountID . pack("v2 C5 a20", 0x82d, $len + 29,$totalchars,0,0,0,$totalchars,-0); # 29 = v2 C5 a20 size for bRO
+	} elsif ($self->{type}->{$config{server_type}}->{charListPacket} eq '0x99d') {		
+		$data = pack("C*", 0x9D, 0x09) .
+		pack("v", $len + 4);
 	} else {
 		$data = $accountID . pack("v v C3", 0x6b, $len + 7, $totalslots, -1, -1);
 	}
 	
 	# Character Block
 	my $block;
 	
-	# Filling Character 1 Block
-	$cID = $charID;	$hp = 10000; $maxHp = 10000; $sp = 10000; $maxSp = 10000; $hairstyle = 1; $level = 99; $headTop = 0; $hairColor = 6;
-	$name = "Poseidon"; $str = 1; $agi = 1; $vit = 1; $int = 1; $dex = 1; $luk = 1;	$exp = 1; $zeny = 1; $jobExp = 1; $jobLevel = 50; $slot = 0; $rename = 0;
-	
-	# Preparing Character 1 Block
-	$block = pack($packstring,$cID,$exp,$zeny,$jobExp,$jobLevel,$opt1,$opt2,$option,$stance,$manner,$statpt,$hp,$maxHp,$sp,$maxSp,$walkspeed,$jobId,$hairstyle,$weapon,$level,$skillpt,$headLow,$shield,$headTop,$headMid,$hairColor,$clothesColor,$name,$str,$agi,$vit,$int,$dex,$luk,$slot,$rename);
+	if ($self->{type}->{$config{server_type}}->{charListPacket} eq '0x99d') {
```

Comment:
Please review this. It seems to me there's a lot of duplicated code in this if-else